BuildResidentialHPXML/measure.rb

```diff
@@ -4142,33 +4165,59 @@ def self.set_hvac_distribution(hpxml, runner, args)
         end
       end
     end
-    return unless air_distribution_systems.size > 0
+
+    # FanCoil?
+    fan_coil_distribution_systems = []
+    hpxml.heating_systems.each do |heating_system|
+      if args[:heating_system_type].include?('Fan Coil')
+        fan_coil_distribution_systems << heating_system
+      end
+    end
+
+    return if air_distribution_systems.size == 0 && fan_coil_distribution_systems.size == 0
 
     if args[:ducts_number_of_return_registers] != Constants.Auto
       number_of_return_registers = args[:ducts_number_of_return_registers]
     end
 
-    hpxml.hvac_distributions.add(id: 'AirDistribution',
-                                 distribution_system_type: HPXML::HVACDistributionTypeAir,
-                                 conditioned_floor_area_served: args[:geometry_cfa],
-                                 number_of_return_registers: number_of_return_registers)
-
-    air_distribution_systems.each do |hvac_system|
-      hvac_system.distribution_system_idref = hpxml.hvac_distributions[-1].id
+    if air_distribution_systems.size > 0
+      hpxml.hvac_distributions.add(id: 'AirDistribution',
+                                   distribution_system_type: HPXML::HVACDistributionTypeAir,
+                                   conditioned_floor_area_served: args[:geometry_cfa],
+                                   number_of_return_registers: number_of_return_registers)
+      air_distribution_systems.each do |hvac_system|
+        hvac_system.distribution_system_idref = hpxml.hvac_distributions[-1].id
+      end
+      set_duct_leakages(args, hpxml.hvac_distributions[-1])
+      set_ducts(args, hpxml.hvac_distributions[-1])
+    end
+
+    if fan_coil_distribution_systems.size > 0
+      hpxml.hvac_distributions.add(id: 'FanCoilDistribution',
+                                   distribution_system_type: HPXML::HVACDistributionTypeAir,
+                                   conditioned_floor_area_served: args[:geometry_cfa],
```

Comment:
I'm not immediately sure how to handle it, but if we have a shared boiler w/ fan coil plus a 2nd heating system, the sum of their conditioned_floor_area_served is going to be greater than the building CFA and result in an error.

Reply:
Although maybe we should relax that constraint... 🤔